core/deployment/src/main/java/io/quarkus/deployment/ide/IdeProcessor.java

```diff
@@ -24,15 +25,37 @@
 
     private static Map<String, List<Ide>> IDE_MARKER_FILES = new HashMap<>();
     private static Map<Predicate<ProcessInfo>, Ide> IDE_PROCESSES = new HashMap<>();
+    private static Map<Ide, Function<ProcessInfo, String>> IDE_ARGUMENTS_EXEC_INDICATOR = new HashMap<>();
 
     static {
         IDE_MARKER_FILES.put(".idea", Collections.singletonList(Ide.IDEA));
         IDE_MARKER_FILES.put(".project", Arrays.asList(Ide.VSCODE, Ide.ECLIPSE));
+        IDE_MARKER_FILES.put("nbactions.xml", Collections.singletonList(Ide.NETBEANS));
+        IDE_MARKER_FILES.put("nb-configuration.xml", Collections.singletonList(Ide.NETBEANS));
+
         IDE_MARKER_FILES = Collections.unmodifiableMap(IDE_MARKER_FILES);
 
-        IDE_PROCESSES.put((processInfo -> processInfo.getCommand().contains("idea")), Ide.IDEA);
-        IDE_PROCESSES.put((processInfo -> processInfo.getCommand().contains("code")), Ide.VSCODE);
-        IDE_PROCESSES.put((processInfo -> processInfo.getCommand().contains("eclipse")), Ide.ECLIPSE); // TODO: I have no idea if this is correct
+        IDE_PROCESSES.put((processInfo -> processInfo.containInCommand("idea")), Ide.IDEA);
+        IDE_PROCESSES.put((processInfo -> processInfo.containInCommand("code")), Ide.VSCODE);
+        IDE_PROCESSES.put((processInfo -> processInfo.containInCommand("eclipse")), Ide.ECLIPSE); // TODO: I have no idea if this is correct
```

Comment:
Can you remove this TODO please while you are at it? It has since been tested :)

Reply:
Done !